assets/helpers/paymentIntegrations/newPaymentFlow/payPalRecurringCheckout.js

```diff
@@ -0,0 +1,136 @@
+// @flow
+
+// ----- Imports ----- //
+
+import { logException } from 'helpers/logger';
+import { routes } from 'helpers/routes';
+import type { Csrf as CsrfState } from 'helpers/csrf/csrfReducer';
+import type { IsoCurrency } from 'helpers/internationalisation/currency';
+
+// ----- Functions ----- //
+
+function loadPayPalRecurring(): Promise<void> {
+  return new Promise((resolve) => {
+    const script = document.createElement('script');
+    script.onload = resolve;
+    script.src = 'https://www.paypalobjects.com/api/checkout.js';
+    if (document.head) {
+      document.head.appendChild(script);
+    }
+  });
+}
+
+// ----- Auxiliary Functions -----//
+
+function payPalRequestData(bodyObj: Object, csrfToken: string) {
+
+  const body = JSON.stringify(bodyObj);
+
+  return {
+    credentials: 'include',
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json', 'Csrf-Token': csrfToken },
+    body,
+  };
+}
+
+function setupPayment(
+  currencyId: IsoCurrency,
+  csrf: CsrfState,
+  setupRecurringPayPalPayment: (Function, Function, IsoCurrency, CsrfState) => void,
+) {
+  return (resolve, reject) => {
+    setupRecurringPayPalPayment(resolve, reject, currencyId, csrf);
+  };
+}
+
+function getPayPalEnvironment(isTestUser: boolean): string {
+  return isTestUser ? window.guardian.payPalEnvironment.uat : window.guardian.payPalEnvironment.default;
+}
+
+function createAgreement(payPalData: Object, csrf: CsrfState) {
+  const body = { token: payPalData.paymentToken };
+  const csrfToken = csrf.token;
+
+  return fetch(routes.payPalCreateAgreement, payPalRequestData(body, csrfToken || ''))
+    .then(response => response.json());
+}
+
+let validateCalled;
```

Comment:
In this case it wouldn't make too much sense. In some sense this information is in Redux, as `payPalHasLoaded`, and is used in `PayPalRecurringButton`. This is really just an extra guard against edge cases so I think it would not be worth the extra wiring to move into Redux